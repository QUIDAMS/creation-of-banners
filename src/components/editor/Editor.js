import React from 'react';
import PropTypes from 'prop-types';

const Editor = ({banner, enterInformationForBanner}) => {
  return(
    <div className='editor'>
      <div className='editor__list'>
        <div className='editor__item'>
          <label className='editor__label' id='1'>Иллюстрация</label>
          <input type='text'
            name='1'
            value={banner.image}
            onChange={e => enterInformationForBanner(e, 'image')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Строка 1</label>
          <input type='text'
            value={banner.line1}
            onChange={e => enterInformationForBanner(e, 'line1')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Строка 2</label>
          <input type='text'
            value={banner.line2}
            onChange={e => enterInformationForBanner(e, 'line2')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Строка 3</label>
          <input type='text'
            value={banner.line3}
            onChange={e => enterInformationForBanner(e, 'line3')}/>
        </div>
        <div className='editor__item'>
        <label className='editor__label'>Ссылка</label>
        <input type='text'
        value={banner.url}
        onChange={e => enterInformationForBanner(e, 'url')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Цвет фона</label>
          <input type='text'
            value={banner.color}
            onChange={e => enterInformationForBanner(e, 'color')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Градиент цвет 1</label>
          <input type='color'
            value={banner.topColorGradient}
            onChange={e => enterInformationForBanner(e, 'topColorGradient')}/>
        </div>
        <div className='editor__item'>
          <label className='editor__label'>Градиент цвет 2</label>
          <input type='color'
            value={banner.bottomColorGradient}
            onChange={e => enterInformationForBanner(e, 'bottomColorGradient')}/>
        </div>
      </div>
    </div>
  );
}

Editor.propTypes = {
  banner: PropTypes.object,
  enterInformationForBanner: PropTypes.func,
};

export default Editor;
