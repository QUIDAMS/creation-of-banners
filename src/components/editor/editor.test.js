import React from 'react';
import Editor from './Editor';
import {shallow} from 'enzyme';

describe('Testing Editor', () => {
  const props = {
    banner: {
      image: '',
      line1: '',
      line2: '',
      line3: '',
      color: '',
      topColorGradient: '',
      bottomColorGradient: '',
      url: '',
    }
  }
  it('rendered correctly', () => {
    const editor = shallow(<Editor banner={props}/>);
    expect(editor).toMatchSnapshot();
  });
});
