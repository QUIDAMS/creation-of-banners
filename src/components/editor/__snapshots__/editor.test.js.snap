// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Editor rendered correctly 1`] = `
<div
  className="editor"
>
  <div
    className="editor__list"
  >
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
        id="1"
      >
        Иллюстрация
      </label>
      <input
        name="1"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Строка 1
      </label>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Строка 2
      </label>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Строка 3
      </label>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Ссылка
      </label>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Цвет фона
      </label>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Градиент цвет 1
      </label>
      <input
        onChange={[Function]}
        type="color"
      />
    </div>
    <div
      className="editor__item"
    >
      <label
        className="editor__label"
      >
        Градиент цвет 2
      </label>
      <input
        onChange={[Function]}
        type="color"
      />
    </div>
  </div>
</div>
`;
